pkg/migration/generate.go

```diff
@@ -467,6 +467,9 @@ func (h *Handler) constructStorageCluster(ds *appsv1.DaemonSet) *corev1.StorageC
 	// Populate StorageCluster spec.volumes from extra volumes mounted to portworx container
 	cluster.Spec.Volumes = getVolumeSpecs(c.VolumeMounts, ds.Spec.Template.Spec.Volumes, builtinPxVolumeNames)
 
+	// Add service type annotation based on portworx and portworx-api service if it's not ClusterIP by default
+	h.addServiceTypeAnnotation(cluster)
```

Comment:
the error should be handled, if there is network error or k8s api server returns some temporary error such as "server busy", this should be returned so we can retry. inside of addServiceTypeAnnotation if service is not found write a log and return nil error, otherwise return error.